ranger/ext/img_display.py

```diff
@@ -388,6 +416,52 @@ def _get_image_dimensions(path):
         return width, height
 
 
+@register_image_displayer("sixel")
+class SixelImageDisplayer(ImageDisplayer, FileManagerAware):
+    """Implementation of ImageDisplayer using SIXEL."""
+
+    def __init__(self):
+        self.win = None
+
+    def draw(self, path, start_x, start_y, width, height):
+        if self.win is None:
+            self.win = self.fm.ui.win.subwin(height, width, start_y, start_x)
+        else:
+            self.win.mvwin(start_y, start_x)
+            self.win.resize(height, width)
+
+        font_width, font_height = get_font_dimensions()
+        fit_width = font_width * width
+        fit_height = font_height * height
+
+        sixel_extra_args = shlex.split(self.fm.settings.sixel_extra_args)
```

Comment:
I'm not a huge fan of this extra_args mechanism tbh. (I'm keeping #2497 in mind here.)
I don't really see a reason to privilege `convert` as the main SIXEL previewer and adding customizability but not quite the functionality from #2497 strikes me as odd.

I think the `sixel` method should either be completely configurable, like the `user` method you're implementing in the other PR pretty much or it shouldn't be at all customizable.

I'm perfectly fine with the latter option since we can add a fully customizable previewer in addition.
However, that still leaves two approaches. Either we pick one SIXEL previewer that we deem best, which can very well be ImageMagick's `convert`, or we mimic the approach that is common across ranger to simply attempt to find something that works. The method would try to find the scripts/binaries for each of the SIXEL previewers we know of and use the first one it encounters in order from rare to common, since if someone has a rare program installed, it's probably because that's their preference.

Let me know what you think?
And *thank you* for working on this. SIXEL previews are a long-awaited feature for many ranger users. 🤗

Reply:
Hi @toonn,

Thank you for responding and sorry for the delay from my side, but here I am.

Just for the record, I actually switched to #2497 and am personally happier with it.

Regarding the extra args mechanism, I opted for it because depending on what kind of
images you have, dithering may or may not be preferred. I now removed the extra_args mechanism
and replaced it with a setting for the preferred dithering method.